File: API/temsAPI/temsAPI/Helpers/Filters/AllocationFilter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using temsAPI.Helpers.Filters.Contracts;

namespace temsAPI.Helpers.Filters
{
    public class AllocationFilter : IPaginationFilter, IEquipmentLabel
    {
        public int Skip { get; set; } = 0;
        public int Take { get; set; } = int.MaxValue;

        public List<string> Personnel { get; set; }
        public List<string> Rooms { get; set; }
        public List<string> Definitions { get; set; }
        public List<string> Equipment { get; set; } 
        public List<string> IncludeLabels { get; set; } // Equipment, Component, Part (Case sensitive)
        public List<string> IncludeStatuses { get; set; } // Active, Closed (Case sensitive)
    }
}

File: New tems/Frontend/Tems/src/app/helpers/filters/allocation.filter.ts
export class AllocationFilter{
    skip: number = 0;
    take: number = 2_147_483_647;

    personnel: string[];
    rooms: string[];
    equipment: string[];
    definitions: string[];
    includeLabels: string[]; // equipment / component / part
    includeStatuses: string[]; // open / closed
}